mon-pix/tests/integration/components/autonomous-course/landing-page-start-block_test.js

```diff
@@ -0,0 +1,115 @@
+/* eslint ember/no-classic-classes: 0 */
+
+import sinon from 'sinon';
+import { module, test } from 'qunit';
+import Service from '@ember/service';
+import { click } from '@ember/test-helpers';
+import { hbs } from 'ember-cli-htmlbars';
+import { render } from '@1024pix/ember-testing-library';
+import setupIntlRenderingTest from '../../../helpers/setup-intl-rendering';
+
+module('Integration | Component | Autonomous Course | Landing page start block', function (hooks) {
+  setupIntlRenderingTest(hooks);
+
+  test('should display title and custom landing page text', async function (assert) {
+    // given
+    this.set('model', {
+      title: 'dummy landing page title',
+      customLandingPageText: 'dummy landing page text',
+    });
+
+    // when
+    const screen = await render(hbs`<AutonomousCourse::LandingPageStartBlock @campaign={{this.model}} />`);
+
+    // then
+    assert
+      .dom(
+        screen.getByText(`${this.intl.t('pages.autonomous-course.landing-page.texts.title')} dummy landing page title`),
+      )
+      .exists();
+    assert.dom(screen.getByText('dummy landing page text')).exists();
+  });
+
+  module('when user is anonymous', function () {
+    test('should display the launcher block', async function (assert) {
+      // when
+      const screen = await render(hbs`<AutonomousCourse::LandingPageStartBlock />`);
+
+      // then
+      assert
+        .dom(
+          screen.getByRole('button', {
+            name: this.intl.t('pages.autonomous-course.landing-page.actions.start-anonymously'),
+          }),
+        )
+        .exists();
+    });
+
+    test('should start campaign participation on main button click', async function (assert) {
+      // given
+      this.set('startCampaignParticipation', sinon.stub());
+
+      // when
+      const screen = await render(
+        hbs`<AutonomousCourse::LandingPageStartBlock @startCampaignParticipation={{this.startCampaignParticipation}} />`,
+      );
+
+      // then
+      await click(
+        screen.getByRole('button', {
+          name: this.intl.t('pages.autonomous-course.landing-page.actions.start-anonymously'),
+        }),
+      );
+      sinon.assert.calledOnce(this.startCampaignParticipation);
+      assert.ok(true);
+    });
+
+    test('should redirect to log-in form on specific button click', async function (assert) {
+      // given
+      class sessionService extends Service {
+        requireAuthenticationAndApprovedTermsOfService = sinon.stub().resolves();
+      }
+      this.owner.register('service:session', sessionService);
```

Comment:
**suggestion** 

const session = this.owner.lookup('service:session') .

Et normalement tu devrais pouvoir stub `requireAuthenticationAndApprovedTermsOfService` sans avoir a balancer ta règle eslint en haut de page ( je suppose qu'ellle est là pour lui )

Reply:
En fait non, c'était pour cet `extend`, mais j'ai pu trouver une autre façon de faire, qui évite de désactiver la règle Eslint.